core/src/stored/reserve.cc

```diff
@@ -641,6 +641,18 @@ static int ReserveDevice(ReserveContext& rctx)
     return -1;
   }
 
+  // Make sure device access mode matches
+  Dmsg3(debuglevel, "chk AccessMode append=%d access_mode=%d\n", rctx.append,
+        rctx.device_resource->access_mode);
+  if (rctx.append && rctx.device_resource->access_mode == IODirection::READ) {
+    // Trying to write but access mode is readonly
+    return -1;
+  } else if (!rctx.append
+             && rctx.device_resource->access_mode == IODirection::WRITE) {
+    // Trying to read but access mode is writeonly
+    return -1;
+  }
```

Comment:
if I understand correctly, then IODirection::NONE and IODirection::READ_WRITE will allow reading and writing here, correct?

Reply:
I guess a configuration setting NONE here should be considered malformed.